lib/widgets/checkbox_widget.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:todo_app/screens/detail_page.dart';
+import 'package:todo_app/src/constants.dart';
+import 'package:todo_app/src/models/note.dart';
+import 'package:todo_app/src/provider/provider_data.dart';
+import 'package:todo_app/src/services.dart';
+
+///
+class CheckBoxWidget extends StatefulWidget {
+  ///
+  final Note note;
+
+  ///
+  const CheckBoxWidget({
+    Key? key,
+    required this.note,
+  }) : super(key: key);
+
+  @override
+  State<CheckBoxWidget> createState() => _CheckBoxState();
+}
+
+class _CheckBoxState extends State<CheckBoxWidget> {
+  List<Todo> taskList = [];
```

Comment:
Make sure all fields under `State` are private.

This way we're closing a very nasty shortcut where a parent widget could try reading state of its child and meddling with it.
A good rule of thumb is -- if there's no explicit reason to make a field/function public -- keep it private.
```suggestion
  List<Todo> _taskList = [];
```